src/init.cpp

```diff
@@ -1503,6 +1503,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     if (GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION))
         StartTorControl(threadGroup, scheduler);
 
+    SetNodeStartHeight(chainActive.Height());
```

Comment:
Should IMO be prefixed Net in some way, otherwise it's not clear that this is a net function.

Also: this creates a closely-coupled binding between Main and Net - just the other way around - my proposal would be to add a node notification signal on changed height that gets passed the height (if such doesn't exist yet), and bind that to SetNetNodeHeight().